### PR TITLE
[#2151] Update Stylelint

### PR DESCRIPTION
**Pull request by sopa301**
Part of #2151.

### Proposed commit message

```
Update Stylelint

The depreciated rules have been moved to a dedicated plugin. We can now
update our Stylelint linter to the latest version.

Let's do so, as well as update related dependencies.
```

### Other information

This was done in a separate PR because:
1. It involves breaking changes
2. It replaces a large part of the Stylelint installation

There's a few new rules that the updated Stylelint has added. I have disabled them for now, and we should discuss which ones to enable. Here's the list:
- `shorthand-property-no-redundant-values`
- `declaration-empty-line-before`
- `selector-class-pattern`
- `color-function-notation`
- `alpha-value-notation`
- `comment-empty-line-before`
- `property-no-vendor-prefix`
- `selector-pseudo-element-colon-notation`
- `at-rule-empty-line-before`
- `font-family-name-quotes`
- `color-hex-length`
- `scss/at-import-no-partial-leading-underscore`
- `scss/at-import-partial-extension`
- `scss/load-no-partial-leading-underscore`
- `scss/double-slash-comment-empty-line-before`
- `scss/double-slash-comment-whitespace-inside`
- `scss/dollar-variable-pattern`

### CODE REVIEW

**Review by asdfghjkxd — APPROVED** [review]
LGTM! Good work on the updating of NodeJS to version 18!

File: frontend/.stylelintrc.json [hunk 1]
```diff
@@ -1,43 +1,60 @@
 {
   "plugins": [
+    "@stylistic/stylelint-plugin",
     "stylelint-order"
   ],
   "extends": [
-    "stylelint-config-recommended-scss",
+    "stylelint-config-standard",
+    "stylelint-config-standard-scss",
```
Comment (vvidday) [inline frontend/.stylelintrc.json:8]: Can I check why we are moving from the recommended config to the standard + standard scss config?

**Review by vvidday — COMMENTED** [review]
Thanks for taking this on @sopa301, definitely agree with disabling the new rules for now and leave those up for discussion (feel free to file a separate issue for that)

File: frontend/.stylelintrc.json (hunk 1 above)
Comment (sopa301) [reply frontend/.stylelintrc.json:8]: `standard` was suggested by the [15.0.0 migration guide](https://stylelint.io/migration-guide/to-15) and `standard-scss` was suggested by the `recommended-scss` [github](https://github.com/stylelint-scss/stylelint-config-recommended-scss).

Do let me know if you want to keep it at the old configs!

Comment (vvidday) [reply frontend/.stylelintrc.json:8]: Got it, thanks! Let's go with these configs.

**Review by vvidday — APPROVED** [review]
LGTM, thanks!

Will have to trouble you to fix the conflicts before merging

File: frontend/babel.config.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 module.exports = {
   presets: [
     '@vue/cli-plugin-babel/preset',
-  ],
-};
+    ],
+}
```
Comment (ckcherry23) [inline frontend/babel.config.js:5]: I think this should be restored?

**Review by ckcherry23 — APPROVED** [review]
LGTM!

**Pull request merged**